src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator/MemberInfoExtensions.cs

```diff
@@ -2,16 +2,18 @@
 using System.Linq;
 using System.Reflection;
 using Microsoft.AspNetCore.Mvc;
-
+using System;
 namespace Swashbuckle.AspNetCore.SwaggerGen
 {
     public static class MemberInfoExtensions
     {
         public static IEnumerable<object> GetInlineOrMetadataTypeAttributes(this MemberInfo memberInfo)
         {
+            if (memberInfo==null) throw new NullReferenceException(nameof(memberInfo));
             var metadataTypeAttribute = memberInfo.DeclaringType.GetCustomAttributes(true)
                 .OfType<ModelMetadataTypeAttribute>()
                 .FirstOrDefault();
```

Comment:
If `DeclaringType` is null, then the `ModelMetadataTypeAttribute` simply isn't applicable and so we should stick to returning the original member attributes rather than throwing an exception. I think the only code change neccessary for this is to postfix the reference to the `DeclaringType` property above with the safe navigation operator (i.e "?").

Reply:
Or return `Enumerable.Empty<object>()` ?

Reply:
I believe the fallback call to `memberInfo.GetCustomAttributes(true)` below will do that automatically